lib/puppet/provider/exec/powershell.rb

```diff
@@ -68,7 +68,8 @@ def run(command, check = false)
         return super("cmd.exe /c \"\"#{native_path(command(:powershell))}\" #{legacy_args} -Command - < \"#{native_path}\"\"", check)
       end
     else
-      result = ps_manager.execute(command)
+      timeout_ms = resource[:timeout].nil? ? nil : resource[:timeout].to_i * 1000
```

Comment:
Powershell ultimately needs an integer.  Althought I believe you're right about precision.  Perhaps;

```
(resource[:timeout] * 1000).to_i
```